src/util/usePaginatedQuery.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2024-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { useEffect, useState } from "react";
+import useDebounce from "./useDebounce";
+
+interface UseDelayedQuery {
+  defaultQuery?: string;
+  defaultPage?: number;
+  debounceTime?: number;
+}
+
+export const usePaginatedQuery = ({ defaultQuery = "", defaultPage = 1, debounceTime = 300 }: UseDelayedQuery = {}) => {
+  const [query, _setQuery] = useState(defaultQuery ?? "");
```

Comment:
Kan ikke denne bare hete `setQuery`?

Reply:
Jo, tenkte bare å legge til rette for eventuelle side-effects senere.

Reply:
Kan vi ikke bare endre det når det legges til isåfall?